tests/test_overrides.py

```diff
@@ -53,17 +53,17 @@ def test_list_page_objects_all():
     # registry's @handle_urls annotation was used.
     assert page_objects == POS.union({POLibSub})
     for rule in rules:
-        assert rule.instead_of == rule.use.expected_overrides, rule.use
-        assert rule.for_patterns == rule.use.expected_patterns, rule.use
-        assert rule.meta == rule.use.expected_meta, rule.use
+        assert rule.instead_of == rule.use.expected_overrides, rule.use  # type: ignore[attr-defined]
```

Comment:
Why is it failing, is a missing type annotation somewhere? I.e. why isn't mypy aware that rules contains Rule instances?

Reply:
The exact error here is `"Callable[..., Any]" has no attribute "expected_overrides"  [attr-defined]` which is mainly due to the test definitions in https://github.com/scrapinghub/web-poet/blob/master/tests/po_lib/__init__.py#L15-L17.

Speaking of this error, I realized that we're labelling POs as `Callable`. I've updated the code to make it `Type[ItemPage]` instead. https://github.com/scrapinghub/web-poet/pull/78/commits/ceab08e855f46dcfa97a79974139680e4c071eec This ensures that we're promoting the overrides to be a subclass of `ItemPage`. I'm also not quite sure why we were labelling POs as `Callable` before. 🤔  What do you think of this change?

The `type: ignore` is still needed here since it's expecting `Type[ItemPage]` which doesn't contain class attributes like `expected_overrides` which are only used in the tests.